### PR TITLE
Alert message popup

### PR DESCRIPTION
**Pull request by h-qadeer**
### Goal
Create Pop-up message

### Work
Create Pop-up message

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,17 +4,23 @@
...
     "@material-ui/icons": "^4.11.2",
     "@material-ui/lab": "^4.0.0-alpha.60",
     "@material-ui/pickers": "^3.3.10",
+    "@mui/icons-material": "^5.0.1",
     "cra-template": "1.1.2",
     "date-fns": "^2.24.0",
     "history": "^5.0.1",
     "json-server": "^0.16.3",
     "lodash": "^4.17.21",
+    "notistack": "^2.0.2",
```
Comment (SiddiquiMonis401) [inline package.json:20]: remove these packages

File: src/components/AlertMessage/index.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    return () => {
+      clearTimeout(TIMER);
+    };
+  }, [SHOW, TIMER]);
+
+  return (
+    SHOW && (
+      <Container time={TIME}>
+        <CheckCircle style={{ fill: 'green' }} />
+        <SuccessMessage>{MESSAGE}</SuccessMessage>
```
Comment (SiddiquiMonis401) [inline src/components/AlertMessage/index.js:43]: Is the message always going to be the success message ?

File: src/components/AlertMessage/style.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+  border: transparent;
+  background-color: white;
+  border-left: 10px solid green;
+  color: black;
+  box-shadow: 0 3px 6px rgba(0, 0, 0, 0.16), 0 3px 6px rgba(0, 0, 0, 0.23);
+  display: flex;
+  justify-content: center;
+  align-items: center;
+
+  animation: ${fadein} 0.5s, ${fadeout} 0.5s ${(props) => props.time};
```
Comment (SiddiquiMonis401) [inline src/components/AlertMessage/style.js:51]: we have built in animations in Material go through them

**Review by SiddiquiMonis401 — APPROVED** [review]

**Review by SiddiquiMonis401 — CHANGES_REQUESTED** [review]
please remove all consoles

File: package.json [hunk 4]
```diff
@@ -4,17 +4,22 @@
   "private": true,
   "dependencies": {
     "@date-io/date-fns": "^1.3.13",
+    "@emotion/react": "^11.4.1",
+    "@emotion/styled": "^11.3.0",
```
Comment (SiddiquiMonis401) [inline package.json:8]: Please remove those dependicies which are not in use

File: src/components/AlertMessage/alertRedux/reduc.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
+const initialState = {
+  toggleSnackbar: false,
+  snackbarMessage: '',
+};
+
+export default function uiReducer(state = initialState, action) {
+  switch (action.type) {
+    case 'TOGGLE_SNACKBAR_OPEN': {
```
Comment (SiddiquiMonis401) [inline src/components/AlertMessage/alertRedux/reduc.js:8]: Please use constants here !!

File: src/components/AlertMessage/index.js [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+import { useDispatch, useSelector } from 'react-redux';
+
+import { toggleSnackbarClose } from './alertRedux/actions';
+import { Container, CrossIcon, SuccessMessage, BTN, SuccessIconButton, ErrorIconButton } from './style';
+
+const Snackbar = ({ timeout, type }) => {
+  const dispatch = useDispatch();
+  const {
+    uiReducer: { toggleSnackbar: SHOW, snackbarMessage: MESSAGE },
+  } = useSelector((state) => state);
```
Comment (SiddiquiMonis401) [inline src/components/AlertMessage/index.js:12]: This is not the right way to write reducer for useSelector please have a look at other useSelectors..

**Review by SiddiquiMonis401 — CHANGES_REQUESTED** [review]

**Review by SiddiquiMonis401 — APPROVED** [review]

**Review by ShahirZain — APPROVED** [review]

**Pull request merged**